treeherder/log_parser/utils.py

```diff
@@ -243,70 +243,73 @@ def post_log_artifacts(project,
                        extract_artifacts_cb,
                        check_errors=False):
     """Post a list of artifacts to a job."""
-    try:
-        credentials = OAuthCredentials.get_credentials(project)
-        req = TreeherderRequest(
-            protocol=settings.TREEHERDER_REQUEST_PROTOCOL,
-            host=settings.TREEHERDER_REQUEST_HOST,
-            project=project,
-            oauth_key=credentials.get('consumer_key', None),
-            oauth_secret=credentials.get('consumer_secret', None),
-        )
-        update_endpoint = 'job-log-url/{0}/update_parse_status'.format(
-            job_log_url['id']
-        )
-
-        logger.debug("Downloading and extracting log information for guid "
-                     "'%s' (from %s)" % (job_guid, job_log_url['url']))
+    def _retry():
+        # Initially retry after 1 minute, then for each subsequent retry
+        # lengthen the retry time by another minute.
+        retry_task.retry(exc=e, countdown=(1 + retry_task.request.retries) * 60)
+        # .retry() raises a RetryTaskError exception,
+        # so nothing after this function will be executed
 
-        artifact_list = extract_artifacts_cb(job_log_url['url'],
-                                             job_guid, check_errors)
-        # store the artifacts generated
-        tac = TreeherderArtifactCollection()
-        for artifact in artifact_list:
-            ta = tac.get_artifact({
-                "job_guid": artifact[0],
-                "name": artifact[1],
-                "type": artifact[2],
-                "blob": artifact[3]
-            })
-            tac.add(ta)
+    credentials = OAuthCredentials.get_credentials(project)
+    update_endpoint = 'job-log-url/{0}/update_parse_status'.format(
+        job_log_url['id']
+    )
 
-        logger.debug("Finished downloading and processing artifact for guid "
-                     "'%s'" % job_guid)
+    logger.debug("Downloading and extracting log information for guid "
+                 "'%s' (from %s)" % (job_guid, job_log_url['url']))
 
-        req.post(tac)
+    req = TreeherderRequest(
+        protocol=settings.TREEHERDER_REQUEST_PROTOCOL,
+        host=settings.TREEHERDER_REQUEST_HOST,
+        project=project,
+        oauth_key=credentials.get('consumer_key', None),
+        oauth_secret=credentials.get('consumer_secret', None),
+    )
 
-        # send an update to job_log_url
-        # the job_log_url status changes from pending to parsed
+    try:
+        artifact_list = extract_artifacts_cb(job_log_url['url'],
+                                             job_guid, check_errors)
+    except:
```

Comment:
You may want to get an instance of the exception here to pass it to the retry function